crates/database/src/orders.rs

```diff
@@ -445,6 +449,61 @@ AND cancellation_timestamp IS NULL
         .map(|_| ())
 }
 
+/// One row in the `order_quote_interactions` table.
+#[derive(Clone, Default, Debug, PartialEq, sqlx::FromRow)]
+pub struct OrderQuoteInteraction {
+    pub order_uid: OrderUid,
+    pub index: i32,
+    pub target: Address,
+    pub value: BigDecimal,
+    pub call_data: Vec<u8>,
+}
+
+pub async fn insert_order_quote_interaction(
+    ex: &mut PgConnection,
+    quote_interaction: &OrderQuoteInteraction,
+) -> Result<(), sqlx::Error> {
+    const INSERT_ORDER_QUOTE_INTERACTION_QUERY: &str = r#"
+    INSERT INTO order_quote_interactions (
+        order_uid,
+        index,
+        target,
+        value,
+        call_data
+    )
+    VALUES ($1, $2, $3, $4, $5)
+    ON CONFLICT (order_uid, index) DO UPDATE SET
```

Comment:
Inserting a duplicated quote leads to an error, which signals inconsistency. Why do we want to change this behavior for interactions?